frontend/src/main/scala/bloop/bsp/BloopBspServices.scala

```diff
@@ -525,37 +526,65 @@ final class BloopBspServices(
   }
 
   def run(params: bsp.RunParams): BspEndpointResponse[bsp.RunResult] = {
+    def getMainClass(project: Project, state: State): Either[Exception, bsp.ScalaMainClass] = {
```

Comment:
Nit: maybe a more appropriate name would be `parseMainClass` or `readMainClass`? It makes it clearer the main class is coming from the request parameters.